src/redux/thunks/thumb.ts

```diff
@@ -0,0 +1,50 @@
+import { AppThunk } from "../store.ts";
+import { getThumbExts } from "../../api/api.ts";
+
+// --- Cached supported thumbnail extensions helpers ---
+let __thumbExtsCache: Set<string> | null | undefined = undefined; // undefined: not fetched, null: unknown/fallback
```

Comment:
This whole file does not need to exist.

For cache layer for thumb exts, you can reuse https://github.com/cloudreve/frontend/blob/35961604a187a49591fa57a50de8c0dad4bb5b78/src/redux/siteConfigSlice.ts#L27
It has a cache status field `loaded`

`isThumbExtSupportedSync` is redundant. Use https://github.com/cloudreve/frontend/blob/35961604a187a49591fa57a50de8c0dad4bb5b78/src/util/index.ts#L87 and `xxx.indexOf` is sufficent.